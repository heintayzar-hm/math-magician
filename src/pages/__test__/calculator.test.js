import React from 'react';
import { render } from '@testing-library/react';
import CalculatorPage from '../calculator';

describe('CalculatorPage', () => {
  it('renders correctly', () => {
    // Arrange

    // Act
    const { container } = render(<CalculatorPage />);

    // Assert
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the Calculator component', () => {
    // Arrange

    // Act
    const { container } = render(<CalculatorPage />);
    const calculator = container.querySelector('.calculator');

    // Assert
    expect(calculator).not.toBeNull();
  });
});
