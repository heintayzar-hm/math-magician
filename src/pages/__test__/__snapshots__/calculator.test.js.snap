// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorPage renders correctly 1`] = `
<main
  style="display: grid; grid-template-columns: 50% 50%; padding: 30px;"
>
  <h2
    class="calculator"
    style="font-size: 25px; font-weight: 700;"
  >
    Let's Do Some Math,
  </h2>
  <div>
    <section
      class="w-full"
      id="calculator"
    >
      <div
        class="w-full leading-[80px] bg-zinc-500 text-white text-right text-3xl"
        data-testid="output"
      >
        0
      </div>
      <div
        class="grid grid-cols-4  leading-[80px] text-center bg-slate-200"
      >
        <div>
          <button
            class="cursor-pointer border-[1px] border-double border-gray-400"
            data-testid="AC"
            type="button"
          >
            AC
          </button>
          <button
            class="cursor-pointer border-[1px] border-double border-gray-400"
            data-testid="+/-"
            type="button"
          >
            +/-
          </button>
          <button
            class="cursor-pointer border-[1px] border-double border-gray-400"
            data-testid="%"
            type="button"
          >
            %
          </button>
          <button
            class="cursor-pointer bg-yellow-600 border-[1px] border-double border-gray-400"
            data-testid="รท"
            type="button"
          >
            รท
          </button>
        </div>
        <div>
          <button
            class="cursor-pointer border-[1px] border-double border-gray-400"
            data-testid="7"
            type="button"
          >
            7
          </button>
          <button
            class="cursor-pointer border-[1px] border-double border-gray-400"
            data-testid="8"
            type="button"
          >
            8
          </button>
          <button
            class="cursor-pointer border-[1px] border-double border-gray-400"
            data-testid="9"
            type="button"
          >
            9
          </button>
          <button
            class="cursor-pointer bg-yellow-600 border-[1px] border-double border-gray-400"
            data-testid="x"
            type="button"
          >
            x
          </button>
        </div>
        <div>
          <button
            class="cursor-pointer border-[1px] border-double border-gray-400"
            data-testid="4"
            type="button"
          >
            4
          </button>
          <button
            class="cursor-pointer border-[1px] border-double border-gray-400"
            data-testid="5"
            type="button"
          >
            5
          </button>
          <button
            class="cursor-pointer border-[1px] border-double border-gray-400"
            data-testid="6"
            type="button"
          >
            6
          </button>
          <button
            class="cursor-pointer bg-yellow-600 border-[1px] border-double border-gray-400"
            data-testid="-"
            type="button"
          >
            -
          </button>
        </div>
        <div>
          <button
            class="cursor-pointer border-[1px] border-double border-gray-400"
            data-testid="1"
            type="button"
          >
            1
          </button>
          <button
            class="cursor-pointer border-[1px] border-double border-gray-400"
            data-testid="2"
            type="button"
          >
            2
          </button>
          <button
            class="cursor-pointer border-[1px] border-double border-gray-400"
            data-testid="3"
            type="button"
          >
            3
          </button>
          <button
            class="cursor-pointer bg-yellow-600 border-[1px] border-double border-gray-400"
            data-testid="+"
            type="button"
          >
            +
          </button>
        </div>
        <div>
          <button
            class="cursor-pointer col-span-2 text-center border-[1px] border-double border-gray-400"
            data-testid="0"
            type="button"
          >
            0
          </button>
          <button
            class="cursor-pointer border-[1px] border-double border-gray-400"
            data-testid="."
            type="button"
          >
            .
          </button>
          <button
            class="cursor-pointer bg-yellow-600 border-[1px] border-double border-gray-400"
            data-testid="="
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </section>
  </div>
</main>
`;
